### ISSUE TITLE
Exclude aria-live that are direct descendants of body from aria-hidden while modal is open

### ISSUE THREAD

**Issue by PhilippMDoerner** [ng-bootstrap/ng-bootstrap#4851]
I use both CDK and NG-Bootstrap in my side-project.
While I was building a modal within which an interaction happens for which I wanted to use CDK-LiveAnnouncer I noticed an interesting interaction:

Ng-Bootstrap, when opening a modal, appears to be applying aria-hidden="true" to ever direct child of the body-tag as long as it's not a script-tag. This includes the aria-live region that i.e. cdk-live-announcer uses to announce its code.

This of course makes it effectively impossible to announce anything with LiveAnnouncer while an Ng-Bootstrap modal is open.

I would therefore propose that when fetching all elements to apply aria-hidden to when a modal opens, that elements that contain the aria-live attribute should be excluded from that step.

Minimal reproduction to demonstrate the issue: https://stackblitz.com/edit/stackblitz-starters-dpmg6qns?file=src%2Fdialog.ts

**Comment by 4javier** [comment]
Not sure if something else other than the modal itself being "non-silenced" complies with the spec for a modal dialog, but if that's the case, probably just adding a check for `aria-live` to the condition of the inner if statement in this method would solve the issue
```ts
if (sibling !== element && sibling.nodeName !== 'SCRIPT' && sibling.getAttribute('aria-live') !== 'true') { 
```
https://github.com/ng-bootstrap/ng-bootstrap/blob/b117049a0f46998703c296150a03e7b14a1ca6d9/src/modal/modal-stack.ts#L230-L242

**Comment by PhilippMDoerner** [comment]
The spec should not be a problem, as the spec does not require aria-hidden (anymore): https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/

> The aria-modal property introduced by ARIA 1.1 replaces [aria-hidden](https://w3c.github.io/aria/#aria-hidden) for informing assistive technologies that content outside a dialog is inert. However, in legacy dialog implementations where aria-hidden is used to make content outside a dialog inert for assistive technology users, it is important that [...]

**Comment by 4javier** [comment]
But [aria-modal](https://w3c.github.io/aria/#aria-modal) looks like expecting the same "excluding" behaviour for the rest of the page

> When a modal element is displayed, authors SHOULD mark all other contents as inert (such as "inert subtrees" in HTML) if the ability to do so exists in the host language. 
It's just `inert` instead of `aria-hidden`.
